### PR TITLE
cli: some cleanups to the new package `clisqlclient` (3/)

### PR DESCRIPTION
**Pull request by knz**
First 6 commits are from #67457 and prior.
Please ignore those (they will disappear once the prefix PR is merged)

Only the last 3 commits belong to this PR..

This is part of a family of PRs that aim to encapsulate CLI functionality into finer grained packages.

### CODE REVIEW

File: pkg/cli/clisqlclient/txn_shim.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+// a sql.Txn, but sqlConn predates go1.8's support for multiple result sets and
+// so deals directly with the lib/pq driver. See #14964.
+//
+// TODO(knz): This code is incorrect, see
+// https://github.com/cockroachdb/cockroach/issues/67261
+type sqlTxnShim struct {
+	conn *sqlConn
+}
+
+var _ crdb.Tx = sqlTxnShim{}
```
Comment (otan) [inline pkg/cli/clisqlclient/txn_shim.go:31]: i'm somewhat amused we import all of cockroach-go just to ensure it obeys the interface. maybe we should move this assertion to the test to optimise out this import.

**Review by otan — APPROVED** [review]

File: pkg/cli/clisqlclient/txn_shim.go (hunk 1 above)
Comment (knz) [reply pkg/cli/clisqlclient/txn_shim.go:31]: It's not just that, we actually use `crdb.ExecuteInTx` to implement `(*sqlConn).ExecTxn`.

**Comment by knz** [comment]
TFYR!

bors r=otan

**Comment by knz** [comment]
the bors run is currently flaking on #64866. I hope it gets retried.

**Pull request merged**